compiler/luci/CMakeLists.txt

```diff
@@ -1,3 +1,9 @@
+if (STATIC_LUCI)
```

Comment:
I want to request this `STATIC_LUCI` be explictly defined somewhere like `infra/nncc/CMakeLists.txt`
and do not change characteristics before-after introducing this, and certainly not change characteristics of our backend.
And also a short comment here why we added this.

Reply:
> I want to do same thing with several other libraries (for example loco)

_loco_ and other modules that are used in NPU backend needs review from @parjong